test/test_rabbit_helper.py

```diff
@@ -0,0 +1,43 @@
+from unittest.mock import patch, MagicMock
+
+from pika.exceptions import AMQPConnectionError
+from pytest import raises
+
+from app.rabbit_helper import init_rabbitmq
+from test.helpers.when_then_return import create_when_then_return
+
+RM_RABBIT_URL = 'rabbit_url'
+RM_RABBIT_AMQP = "amqp://user:pa55word@host:0001"
+RM_BINDING_KEY = "test-binding-key"
+RM_RABBIT_QUEUE = "Test.Queue"
+RM_RABBIT_CONNECTION = 'rabbit connection'
+RM_RABBIT_EXCHANGE = "test-exchange"
+RM_RABBIT_QUEUE_ARGS = {'x-dead-letter-exchange': 'case-deadletter-exchange',
+                        'x-dead-letter-routing-key': RM_BINDING_KEY}
+
+
+def test_rabbit_init():
+    with patch('app.rabbit_helper.pika') as mock_pika:
+        connection_mock = MagicMock()
+        mock_pika.URLParameters = create_when_then_return(RM_RABBIT_AMQP, return_value=RM_RABBIT_URL)
+        mock_pika.BlockingConnection = create_when_then_return(RM_RABBIT_URL, return_value=connection_mock)
+
+        channel_mock = MagicMock()
+        connection_mock.channel = create_when_then_return(return_value=channel_mock)
+
+        init_rabbitmq(rabbitmq_amqp=RM_RABBIT_AMQP,
+                      binding_key=RM_BINDING_KEY,
+                      exchange_name=RM_RABBIT_EXCHANGE,
+                      queue_name=RM_RABBIT_QUEUE)
+
+        channel_mock.exchange_declare.assert_called_once_with(exchange=RM_RABBIT_EXCHANGE, exchange_type='direct',
+                                                              durable=True)
+        channel_mock.queue_declareassert_called_once_with(queue=RM_RABBIT_QUEUE, durable=True,
```

Comment:
Assert method not called (wasn't caught because the channel_mock MagicMock went ahead and created the `queue_declareassert_called_once_with` attribute for you).

```suggestion
        channel_mock.queue_declare.assert_called_once_with(queue=RM_RABBIT_QUEUE, durable=True,
```

Though I expect this will now fail as the `RM_RABBIT_QUEUE_ARGS` will differ depending on environment.